python/essex-config/tests/unit/test_config.py

```diff
@@ -164,111 +164,106 @@ def test_nested_alias_for_different_source():
 
-    @config(sources=[source])
     class BasicConfiguration(BaseModel):
         hello: str
 
-    basic_config = BasicConfiguration.load_config()
+    basic_config = load_config(BasicConfiguration, sources=[source])
     assert basic_config.hello == "world"
 
     source.data["hello"] = "new world"
 
-    basic_config = BasicConfiguration.load_config()
+    basic_config = load_config(BasicConfiguration, sources=[source])
     assert basic_config.hello == "world"
 
-    basic_config = BasicConfiguration.load_config(refresh_config=True)
+    basic_config = load_config(
+        BasicConfiguration, sources=[source], refresh_config=True
+    )
     assert basic_config.hello == "new world"
 
 
 def test_basic_default_no_value_config():
-    @config(sources=[MockSource()])
     class BasicConfiguration(BaseModel):
         not_a_value_in_config: str = "hello"
 
-    basic_config = BasicConfiguration.load_config()
+    basic_config = load_config(BasicConfiguration, sources=[MockSource()])
     assert basic_config.not_a_value_in_config == "hello"
 
 
 def test_basic_default_no_value_field_config():
-    @config(sources=[MockSource()])
     class BasicConfiguration(BaseModel):
         not_a_value_in_config: str = Field(default="hello")
 
-    basic_config = BasicConfiguration.load_config()
+    basic_config = load_config(BasicConfiguration, sources=[MockSource()])
     assert basic_config.not_a_value_in_config == "hello"
 
 
 def test_basic_default_no_value_field_factory_config():
-    @config(sources=[MockSource()])
     class BasicConfiguration(BaseModel):
         not_a_value_in_config: str = Field(default_factory=lambda: "hello")
 
-    basic_config = BasicConfiguration.load_config()
+    basic_config = load_config(BasicConfiguration, sources=[MockSource()])
     assert basic_config.not_a_value_in_config == "hello"
 
 
 def test_missing_key():
-    @config(sources=[MockSource()])
     class KeyErrorConfig(BaseModel):
         not_valid_key: str
 
     with pytest.raises(
         ValueError,
         match="Value for not_valid_key is required and not found in any source.",
     ):
-        KeyErrorConfig.load_config()
+        load_config(KeyErrorConfig, sources=[MockSource()])
 
 
 def test_not_required_field():
-    @config(sources=[MockSource()])
     class NotRequiredConfig(BaseModel):
         not_required_key: str | None = None
 
-    assert NotRequiredConfig.load_config().not_required_key is None
+    assert (
+        load_config(NotRequiredConfig, sources=[MockSource()]).not_required_key is None
+    )
 
 
 def test_union_type():
-    @config(sources=[MockSource()])
     class UnionTypeConfig(BaseModel):
         hello: int | str | None = None
         no_prefix_field: int | str | None = None
 
-    assert UnionTypeConfig.load_config().hello == "world"
-    assert UnionTypeConfig.load_config().no_prefix_field == 1
+    config = load_config(UnionTypeConfig, sources=[MockSource()])
+    assert config.hello == "world"
```

Comment:
Should we also assert for types?

Reply:
most of these asserts are for == which includes checking type, for example if we did
`assert config.something == "1"` and `something` is an int it would not pass the assertion